### PR TITLE
[Enhancement] Modsecurity event coverage

### PR DESCRIPTION
**Pull request by WildDogOne**
## What does this PR do?

This PR is aimed at getting better coverage on Modsecurity logs.
Mostly focused on the http request side (response is already covered well)
also some bugfix with missing fields ;)

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/modsecurity/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # newer versions go on top
+
+- version: "1.2.2"
```
Comment (efd6) [inline packages/modsecurity/changelog.yml:3]:
```suggestion
- version: "1.3.0"
```
(enhancements will bump minor)

**Review by efd6 — COMMENTED** [review]
I'm wondering whether it would be valuable to have `modsecurity.request.headers` and `modsecurity.response.headers` both as flattened. This could be guarded by an advanced option to prevent overloading indices.

**Comment by WildDogOne** [comment]
@efd6 I have no actual opinion on this xD
I know I did have some issues with flattened field types in the past, but that was mostly due to changes in the mapping associated to this

**Comment by efd6** [comment]
It is always something that we can add in the future if it becomes needed.

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**